### DISCUSSION TITLE
Download docker.io/renovate/go:1.17.1 through artifactory proxy

### DISCUSSION THREAD

**Discussion by XFNeo** [renovatebot/renovate#11906] [Request Help]
HI we use a renovatebot docker image on our jenkins builder wich has no access to internet (only private network). We usually proxy a public docker repository through our artifactory server. How we can proxy request to get image  docker.io/renovate/go:1.17.1 in dependabot? Or maybe we can disable run of go container?

`DEBUG: Using docker to execute (repository=control-plane, branch=renovate/github.com-golang-protobuf-1.x)
DEBUG: Found version constraint - checking for a compatible image to use (repository=control-plane, branch=renovate/github.com-golang-protobuf-1.x)
       "depName": "docker.io/renovate/go",
       "scheme": "npm",
       "constraint": "^1.17"
DEBUG: Failed to update go.sum (repository=control-plane, branch=renovate/github.com-golang-protobuf-1.x)
       "err": {
         "hostType": "docker",
         "err": {
           "name": "RequestError",
           "code": "ECONNRESET",
           "timings": {
             "start": 1632493426918,
             "socket": 1632493426926,
             "lookup": 1632493426926,
             "connect": 1632493426926,
             "error": 1632493426928,
             "phases": {"wait": 8, "dns": 0, "tcp": 0, "total": 10}
           },
           "message": "Client network socket disconnected before secure TLS connection was established",
           "stack": "RequestError: Client network socket disconnected before secure TLS connection was established\n    at ClientRequest.<anonymous> (/usr/src/app/node_modules/got/dist/source/core/index.js:956:111)\n    at Object.onceWrapper (events.js:520:26)\n    at ClientRequest.emit (events.js:412:35)\n    at ClientRequest.emit (domain.js:470:12)\n    at ClientRequest.origin.emit (/usr/src/app/node_modules/@szmarczak/http-timer/dist/source/index.js:43:20)\n    at TLSSocket.socketErrorListener (_http_client.js:475:9)\n    at TLSSocket.emit (events.js:412:35)\n    at TLSSocket.emit (domain.js:470:12)\n    at emitErrorNT (internal/streams/destroy.js:106:8)\n    at emitErrorCloseNT (internal/streams/destroy.js:74:3)\n    at connResetException (internal/errors.js:628:14)\n    at TLSSocket.onConnectEnd (_tls_wrap.js:1569:19)\n    at TLSSocket.emit (events.js:412:35)\n    at TLSSocket.emit (domain.js:470:12)\n    at endReadableNT (internal/streams/readable.js:1317:12)\n    at processTicksAndRejections (internal/process/task_queues.js:82:21)",
           "options": {
             "headers": {
               "user-agent": "RenovateBot/27.19.0 (https://github.com/renovatebot/renovate)",
               "accept-encoding": "gzip, deflate, br"
             },
             "url": "https://index.docker.io/v2/",
             "hostType": "docker",
             "username": "",
             "password": "",
             "method": "GET",
             "http2": false
           }
         },
         "lookupName": "docker.io/renovate/go",
         "message": "external-host-error",
         "stack": "Error: external-host-error\n    at getAuthHeaders (/usr/src/app/node_modules/renovate/lib/datasource/docker/common.ts:182:15)\n    at processTicksAndRejections (internal/process/task_queues.js:95:5)\n    at getDockerApiTags (/usr/src/app/node_modules/renovate/lib/datasource/docker/index.ts:72:19)\n    at getTags (/usr/src/app/node_modules/renovate/lib/datasource/docker/index.ts:114:14)\n    at Object.getReleases (/usr/src/app/node_modules/renovate/lib/datasource/docker/index.ts:255:16)\n    at getRegistryReleases (/usr/src/app/node_modules/renovate/lib/datasource/index.ts:70:15)\n    at fetchReleases (/usr/src/app/node_modules/renovate/lib/datasource/index.ts:227:15)\n    at getPkgReleases (/usr/src/app/node_modules/renovate/lib/datasource/index.ts:285:7)\n    at getDockerTag (/usr/src/app/node_modules/renovate/lib/util/exec/docker/index.ts:86:25)\n    at generateDockerCommand (/usr/src/app/node_modules/renovate/lib/util/exec/docker/index.ts:237:11)\n    at prepareRawExec (/usr/src/app/node_modules/renovate/lib/util/exec/index.ts:117:27)\n    at exec (/usr/src/app/node_modules/renovate/lib/util/exec/index.ts:134:39)\n    at updateArtifacts (/usr/src/app/node_modules/renovate/lib/manager/gomod/artifacts.ts:185:5)\n    at getUpdatedPackageFiles (/usr/src/app/node_modules/renovate/lib/workers/branch/get-updated.ts:198:23)\n    at processBranch (/usr/src/app/node_modules/renovate/lib/workers/branch/index.ts:351:17)\n    at writeUpdates (/usr/src/app/node_modules/renovate/lib/workers/repository/process/write.ts:38:17)\n    at update (/usr/src/app/node_modules/renovate/lib/workers/repository/process/extract-update.ts:129:11)\n    at Object.renovateRepository (/usr/src/app/node_modules/renovate/lib/workers/repository/index.ts:50:17)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:122:7)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       }`

**Comment by HonkingGoose** [comment]
Please format your logging output properly, what you have right now is hard to read.

**Comment by rarkins** [comment]
https://docs.renovatebot.com/self-hosted-configuration/#dockerimageprefix

**Reply by XFNeo** [reply]
Thanks! It works.
